command/agent/config_parse_test.go

```diff
@@ -212,111 +265,13 @@ var basicConfig = &Config{
-		ClientHTTPCheckName:  "nomad-client-http-health-check",
-		Addr:                 "127.0.0.1:9500",
-		AllowUnauthenticated: &trueValue,
-		Token:                "token1",
-		Auth:                 "username:pass",
-		EnableSSL:            &trueValue,
-		VerifySSL:            &trueValue,
-		CAFile:               "/path/to/ca/file",
-		CertFile:             "/path/to/cert/file",
-		KeyFile:              "/path/to/key/file",
-		ServerAutoJoin:       &trueValue,
-		ClientAutoJoin:       &trueValue,
-		AutoAdvertise:        &trueValue,
-		ChecksUseAdvertise:   &trueValue,
-		Timeout:              5 * time.Second,
-		TimeoutHCL:           "5s",
-		UseIdentity:          &trueValue,
-		ServiceIdentity: &config.WorkloadIdentity{
-			Audience: []string{"consul.io", "nomad.dev"},
-			Env:      false,
-			File:     true,
-		},
-		TemplateIdentity: &config.WorkloadIdentity{
-			Audience: []string{"consul.io"},
-			Env:      true,
-			File:     false,
-		},
-	},
+	Consul:                    basicConsulConfig,
 	Consuls: map[string]*config.ConsulConfig{
-		"default": {
-			Name:                 "default",
-			ServerServiceName:    "nomad",
-			ServerHTTPCheckName:  "nomad-server-http-health-check",
-			ServerSerfCheckName:  "nomad-server-serf-health-check",
-			ServerRPCCheckName:   "nomad-server-rpc-health-check",
-			ClientServiceName:    "nomad-client",
-			ClientHTTPCheckName:  "nomad-client-http-health-check",
-			Addr:                 "127.0.0.1:9500",
-			AllowUnauthenticated: &trueValue,
-			Token:                "token1",
-			Auth:                 "username:pass",
-			EnableSSL:            &trueValue,
-			VerifySSL:            &trueValue,
-			CAFile:               "/path/to/ca/file",
-			CertFile:             "/path/to/cert/file",
-			KeyFile:              "/path/to/key/file",
-			ServerAutoJoin:       &trueValue,
-			ClientAutoJoin:       &trueValue,
-			AutoAdvertise:        &trueValue,
-			ChecksUseAdvertise:   &trueValue,
-			Timeout:              5 * time.Second,
-			TimeoutHCL:           "5s",
-			UseIdentity:          &trueValue,
-			ServiceIdentity: &config.WorkloadIdentity{
-				Audience: []string{"consul.io", "nomad.dev"},
-				Env:      false,
-				File:     true,
-			},
-			TemplateIdentity: &config.WorkloadIdentity{
-				Audience: []string{"consul.io"},
-				Env:      true,
-				File:     false,
-			},
-		},
-	},
-	Vault: &config.VaultConfig{
-		Name:                 "default",
-		Addr:                 "127.0.0.1:9500",
-		AllowUnauthenticated: &trueValue,
-		ConnectionRetryIntv:  config.DefaultVaultConnectRetryIntv,
-		Enabled:              &falseValue,
-		Role:                 "test_role",
-		TLSCaFile:            "/path/to/ca/file",
-		TLSCaPath:            "/path/to/ca",
-		TLSCertFile:          "/path/to/cert/file",
-		TLSKeyFile:           "/path/to/key/file",
-		TLSServerName:        "foobar",
-		TLSSkipVerify:        &trueValue,
-		TaskTokenTTL:         "1s",
-		Token:                "12345",
+		"default": basicConsulConfig,
```

Comment:
I'm not sure which change I made that started causing tests to fail unless `default` is the exact same pointer as `Consul`, and same for `Vault`. 

I think this is the structure we want, so maybe I fixed something? 😅